### ISSUE TITLE
[consistent-destructuring] incorrect report on different values shared same identifier name

### ISSUE THREAD

**Issue by JounQin** [sindresorhus/eslint-plugin-unicorn#1109]
```diff
---const { containerStatuses = [], initContainerStatuses = [] } = status
+++const { containerStatuses = [], initContainerStatuses = [], name } = status
const statuses = containerStatuses.concat(...initContainerStatuses)
---return statues.find(status => status.name === container.name)
+++return statuses.find(status => name === container.name)
```

**Comment by fisker** [comment]
Another real world case https://github.com/fisker/eslint-plugin-unicorn/blob/8a3c3ea1984f324e7ac91ba44336e55f8f7bb003/rules/no-static-only-class.js#L77-L95

**Comment by BridgeAR** [comment]
This also came up during my work as an issue. While the main issue is that shadowing is not checked for, it's also an issue in tests in case multiple things are checked. E.g.,

```js
const foobar = 'Some value'
const { error, body } = await rpc('foobar')
assert(!error, error)
assert.deepEqual(foobar, body.foobar)
```

By destructuring the body, it'll fail in case there's an error and `body` is nullish. The actual expected assertion with a proper error message would not be reached and it's more difficult to debug the code.
